### ISSUE TITLE
precision and recall code

### ISSUE THREAD

**Issue by chinmay5** [nyuad-cai/CXR-ML-GZSL#2]
Dear Team,

Thank you so much for the well-structured code. It has been of immense help. I was wondering if it is possible to provide the code block for precision@k, recall@k and f1@k metrics as well. Unfortunately, I was not able to find them in the repo.

**Comment by nasir6** [comment]
from sklearn.metrics import average_precision_score, f1_score, precision_score, recall_score

def compute_F1(predictions_all, labels, k_val, mode_F1=None ):
    f1, p, r = 0,0,0
    idx = predictions_all.topk(dim=1, k=k_val)[1] 
    predictions = torch.zeros_like(predictions_all)
#     predictions.fill_(0)
#     predictions[idx] = 1
    labels = labels.cpu().numpy()
    predictions.scatter_(dim=1,index=idx,src=torch.ones(predictions.size(0),k_val))
    predictions=predictions.cpu().numpy()
    if mode_F1 == 'overall':
        # print('evaluation overall!! cannot decompose into classes F1 score')
        mask = predictions == 1
        TP = (labels[mask] == 1).sum()
#         .float()
        tpfp = mask.sum()
#     .float()
        tpfn = (labels == 1).sum()
#         .float()
        p = TP/ tpfp
        r = TP/tpfnx
        f1 = 2*p*r/(p+r)
    else:
        num_class = predictions.shape[1]
        # print('evaluation per classes')
        f1 = np.zeros(num_class)
        p = np.zeros(num_class)
        r = np.zeros(num_class)
        for idx_cls in range(num_class):
            prediction = np.squeeze(predictions[:, idx_cls])
            label = np.squeeze(labels[:, idx_cls])
            if np.sum(label > 0) == 0:
                continue
            binary_label = np.clip(label, 0, 1)
            f1[idx_cls] = f1_score(binary_label, prediction)
            p[idx_cls] = precision_score(binary_label, prediction)
            r[idx_cls] = recall_score(binary_label, prediction)
    return f1, p, r

**Comment by chinmay5** [comment]
Dear @nasir6 sorry to open the issue again. I was unable to produce the numbers reported by the paper by means of the code you shared. It would be great if you could just integrate the Precision and recall code as part of the repo itself. This would allow for easy reproducibility. Thanks in advance.
